managers/backup-manager/BackupStatusPoller.js

```diff
@@ -0,0 +1,277 @@
+            } else {
+              const patchBody = _.cloneDeep(resourceBody);
+              const metadata = patchBody.metadata;
+              const currentAnnotations = metadata.annotations;
+              const patchAnnotations = currentAnnotations ? currentAnnotations : {};
+              patchAnnotations.lockedByTaskPoller = JSON.stringify({
+                lockTime: new Date(),
+                ip: config.broker_ip
+              });
+              metadata.annotations = patchAnnotations;
+              // Handle conflict also
+              return eventmesh.apiServerClient.updateResource({
+                  resourceGroup: resourceDetails.resourceGroup,
+                  resourceType: resourceDetails.resourceType,
+                  resourceId: metadata.name,
+                  metadata: metadata
+                })
+                .tap(updatedResource => logger.debug(`Successfully acquired backup status poller lock for request with options: ${JSON.stringify(options)}\n` +
+                  `Updated resource with poller annotations is: `, updatedResource))
+                .then(() => {
+                  if (!utils.isServiceFabrikOperationFinished(resourceBody.status.state)) {
+                    const instanceInfo = _.chain(response)
+                      .pick('tenant_id', 'backup_guid', 'instance_guid', 'agent_ip', 'service_id', 'plan_id', 'deployment', 'started_at', 'abortStartTime')
+                      .value();
+                    return BackupStatusPoller.checkOperationCompletionStatus(instanceInfo)
+                      .then(operationStatusResponse => {
+                        if (utils.isServiceFabrikOperationFinished(operationStatusResponse.state)) {
+                          BackupStatusPoller.clearPoller(metadata.name, intervalId);
+                        }
+                      });
+                  } else {
+                    BackupStatusPoller.clearPoller(metadata.name, intervalId);
+                  }
+                })
+                .catch(AssertionError, err => {
+                  logger.error(`Error occured while polling for backup, marking backup as failed`, err);
+                  BackupStatusPoller.clearPoller(metadata.name, intervalId);
+                  return eventmesh.apiServerClient.updateResource({
+                    resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                    resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                    resourceId: metadata.name,
+                    status: {
+                      state: CONST.APISERVER.RESOURCE_STATE.FAILED,
+                      error: utils.buildErrorJson(err)
+                    }
+                  });
+                })
+                .catch(Conflict, () => {
+                  logger.debug(`Not able to acquire backup status poller processing lock for backup with guid ${object.metadata.name}, Request is probably picked by other worker`);
+                });
```

Comment:
What if in case of other error are we not patching the resource ?

Reply:
Not here, this can be a valid error when polling is being done by other process in case of multiple processes running